src/CommandIsolator.cpp

```
@@ -214,17 +228,27 @@ process::Future<::mesos::ResourceStatistics> CommandIsolatorProcess::usage(
 process::Future<Nothing> CommandIsolatorProcess::cleanup(
     const ContainerID& containerId) {
   if (m_cleanupCommand.isNone()) {
+    if (m_infos.contains(containerId)) {
```

here I agree to let cleanup occur naturally even if m_infos does not contain info.

---

Maybe if we fail for watch & usage callback, maybe it's better to also fail for cleanup ?

---

cleanup semantic does not really say if prepare has been called (or actually finished successfully). If we fail in cleanup, it means user have a feedback about an error in cleanup instead of having the real error shown.

---

So maybe just a warning ? If we try to pass containerConfig each time to the cleanup callback, we can't silently ignore errors because user will expect his script to get all the information

---

warning is the right option indeed